fvtr/ck_provides/ck_provides.exp

```diff
@@ -69,27 +69,29 @@ proc process_deb {package expected} {
 	set version "(= ${at_full_ver})"
 
 	if { ![info exists ::env(AT_WD)] } {
-		if { ![exec dpkg-query --status ${expected}]} {
+		if { [catch {exec dpkg-query --status ${expected}}] } {
 			printit "User didn't install package ${expected}.\
 Can't run this test."
 			return ${ENOSYS}
 		}
 
-		return ${ENOSYS}
-
-		# TODO: Adapt this code to the new dummy package.
-		# Keep in mind that, the package installed may not be of the
-		# same version you're trying to test.
-		# The biggest problem is that dpkg-query doesn't provide a way
-		# to query for a specific version of a package.
-		set depends [string trimleft [exec dpkg-query --status \
-						  ${expected} | grep -E "Provides"] "Provides: "]
-		# We need trim here because the query returns the string with
-		# "Provides:" in the begining.
-		set provides [string trimleft [exec dpkg-query --status \
-				${package} | grep -E "Provides"] "Provides: "]
-		if { ![string equal ${expected} ${provides}] } {
-			printit "Provides for ${package} not found."
+		# Verify the version of the dummy package.
+		set dversion [string trimleft [exec dpkg-query --status \
+				${expected} | grep -E "Version"] "Version: "]
+		if { ![string equal ${dversion} ${at_full_ver}] } {
+			printit "Package ${expected} installed is not from AT \
+${at_full_ver}. Can't run this test."
+			return ${ENOSYS}
+		}
+
```

Comment:
It would be nice to see a comment here, something in the lines of:

# Parse the Pre-Depends field from the dummy package.  It must contain the actual package.